nixos/modules/services/web-apps/booklore.nix

```
@@ -0,0 +1,330 @@
+
+      after = [ "network.target" ] ++ lib.optionals cfg.database.createLocally [ "mysql.service" ];
+      wantedBy = [ "multi-user.target" ] ++ lib.optionals cfg.database.createLocally [ "mysql.service" ];
+
+      environment = cfg.environment;
+
+      serviceConfig = {
+        Type = "simple";
+        User = cfg.user;
+        Group = cfg.group;
+        StateDirectory = "booklore";
+        ExecStartPre = "+${startupScript ''
+          echo "ALTER USER '${cfg.database.user}'@'localhost' IDENTIFIED BY '$DATABASE_PASSWORD';" | ${pkgs.mariadb}/bin/mariadb
+        ''}";
+        ExecStart = startupScript (lib.getExe cfg.package);
+
+        Restart = "on-failure";
+        EnvironmentFile = cfg.environmentFile;
+
+        # Hardening
+        CapabilityBoundingSet = "";
+        NoNewPrivileges = true;
+        PrivateUsers = true;
+        PrivateTmp = true;
+        PrivateDevices = true;
+        PrivateMounts = true;
+        ProtectClock = true;
+        ProtectControlGroups = true;
+        ProtectHome = true;
+        ProtectHostname = true;
+        ProtectKernelLogs = true;
+        ProtectKernelModules = true;
+        ProtectKernelTunables = true;
+        RestrictAddressFamilies = [
+          "AF_INET"
+          "AF_INET6"
+          "AF_UNIX"
+        ];
+        RestrictNamespaces = true;
+        RestrictRealtime = true;
+        RestrictSUIDSGID = true;
+      };
+    };
+
+    services.booklore.environment = {
+      BOOKLORE_DATA_DIR = cfg.dataDir;
+      BOOKLORE_BOOKDROP_DIR = cfg.bookdropDir;
+      DATABASE_HOST = if cfg.database.host != null then cfg.database.host else "127.0.0.1";
+      DATABASE_PORT = if cfg.database.port != null then toString cfg.database.port else "3306";
+      DATABASE_USERNAME = cfg.database.user;
+      SERVER_PORT = toString cfg.api.port;
+    };
+
+    services.nginx = lib.mkIf cfg.nginx.enable {
+      enable = true;
+      virtualHosts.${cfg.nginx.virtualHost} = {
+        listen = [
+          {
+            addr = cfg.host;
+            port = cfg.port;
+          }
+        ];
+
+        forceSSL = cfg.nginx.forceSSL;
+        enableACME = lib.mkDefault (cfg.nginx.forceSSL && cfg.nginx.useACMEHost == null);
+        useACMEHost = cfg.nginx.useACMEHost;
+
+        root = "${cfg.package}/share/booklore-ui";
+
+        extraConfig = ''
+          client_max_body_size ${cfg.settings.maxBodySize};
+        '';
+
+        locations = {
+          "/" = {
+            tryFiles = "$uri $uri/ /index.html =404";
+            extraConfig = ''
+              location ~* \.mjs$ {
+                types {
+                    text/javascript mjs;
+                }
+              }
+            '';
+          };
+
+          "/api/" = {
+            proxyPass = "http://localhost:${toString cfg.api.port}";
+            recommendedProxySettings = lib.mkDefault true;
```

If you want Kobo sync to work this should be
```suggestion
            recommendedProxySettings = lib.mkDefault true;
            extraConfig = ''
              proxy_set_header X-Forwarded-Port $server_port;
              proxy_buffer_size 128k;
              proxy_buffers 4 256k;
              proxy_busy_buffers_size 256k;
            '';
```
We also need to add:
```
    services.nginx.appendHttpConfig = ''
      large_client_header_buffers 8 32k; 
    '';
```
according to https://booklore-app.github.io/booklore-docs/docs/integration/kobo/